e2e/src/step_definitions/media-viewer.ts

```diff
@@ -452,3 +456,98 @@ Then('I expect custom toolbar button should be enabled', async () => {
 Then('I expect custom toolbar should be enabled', async () => {
   await page.waitForElement(by.className('customToolbar'));
 });
+
+When(/^The user clicks on the show comments panel toggle icon$/, async function () {
+  await genericMethods.sleep(2000);
+  await commentsPanelPage.clickShowCommentsToggleIcon();
+  await genericMethods.sleep(3000);
+
+  // await commentsPanelPage.clickOnSearchTab();
+
+});
+
+//I expect to see the comments filter and search tabs
+When(/^I expect to be able to click on the Filter Search And Comments Tab$/, async function () {
+  await genericMethods.sleep(2000);
+  await commentsPanelPage.clickOnCommentsTab();
+  await genericMethods.sleep(2000);
+  await commentsPanelPage.clickOnFilterTab();
+  await genericMethods.sleep(2000);
+  await commentsPanelPage.clickOnSearchTab();
+});
+
+
+When(/^I do a invalid search of the Comments Tab$/, async function () {
+  await genericMethods.sleep(2000);
+  await commentsPanelPage.performSearch();
+
+});
+
+
+
+When(/^The user enters a random search string$/, async function () {
```

Comment:
I don't see where the user enters the random search string here

Reply:
Search String is now passed in as a parameter to the performSearch() method.